app/src/ui/dispatcher/dispatcher.ts

```diff
@@ -647,9 +647,20 @@ export class Dispatcher {
     workingDirectory: WorkingDirectoryStatus,
     successfulMergeBannerState: SuccessfulMergeBannerState
   ) {
+    // get manual resolutions in case there are manual conflicts
+    const repositoryState = this.repositoryStateManager.get(repository)
+    const { conflictState } = repositoryState.changesState
+    if (conflictState === null) {
```

Comment:
Did you mean like this?

```diff
diff --git a/app/src/lib/stores/app-store.ts b/app/src/lib/stores/app-store.ts
index ff72a3fee..71821ed17 100644
--- a/app/src/lib/stores/app-store.ts
+++ b/app/src/lib/stores/app-store.ts
@@ -3200,16 +3200,30 @@ export class AppStore extends TypedBaseStore<IAppState> {
   /** This shouldn't be called directly. See `Dispatcher`. */
   public async _finishConflictedMerge(
     repository: Repository,
-    workingDirectory: WorkingDirectoryStatus,
-    manualResolutions: Map<string, ManualConflictResolutionKind>
+    workingDirectory: WorkingDirectoryStatus
   ): Promise<string | undefined> {
+    // get manual resolutions in case there are manual conflicts
+    const repositoryState = this.repositoryStateCache.get(repository)
+    const { conflictState } = repositoryState.changesState
+    if (conflictState === null) {
+      // if this doesn't exist, something is very wrong and we shouldn't proceed 😢
+      log.error(
+        'Conflict state missing during finishConflictedMerge. No merge will be committed.'
+      )
+      return
+    }
+
     // filter out untracked files so we don't commit them
     const trackedFiles = workingDirectory.files.filter(f => {
       return f.status.kind !== AppFileStatusKind.Untracked
     })
     const gitStore = this.gitStoreCache.get(repository)
     return await gitStore.performFailableOperation(() =>
-      createMergeCommit(repository, trackedFiles, manualResolutions)
+      createMergeCommit(
+        repository,
+        trackedFiles,
+        conflictState.manualResolutions
+      )
     )
   }
 
diff --git a/app/src/ui/dispatcher/dispatcher.ts b/app/src/ui/dispatcher/dispatcher.ts
index b2cb4f2ad..106a4b7c4 100644
--- a/app/src/ui/dispatcher/dispatcher.ts
+++ b/app/src/ui/dispatcher/dispatcher.ts
@@ -647,20 +647,9 @@ export class Dispatcher {
     workingDirectory: WorkingDirectoryStatus,
     successfulMergeBannerState: SuccessfulMergeBannerState
   ) {
-    // get manual resolutions in case there are manual conflicts
-    const repositoryState = this.repositoryStateManager.get(repository)
-    const { conflictState } = repositoryState.changesState
-    if (conflictState === null) {
-      // if this doesn't exist, something is very wrong and we shouldn't proceed 😢
-      log.error(
-        'Conflict state missing during finishConflictedMerge. No merge will be committed.'
-      )
-      return
-    }
     const result = await this.appStore._finishConflictedMerge(
       repository,
-      workingDirectory,
-      conflictState.manualResolutions
+      workingDirectory
     )
     if (result !== undefined) {
       this.appStore._setSuccessfulMergeBannerState(successfulMergeBannerState)
```

Reply:
No, i tried that and i'd like to avoid it because it adds some side effects to `_finishConflictedMerge` that make it harder to test (it would then depend on state existing somewhere else in the store). I was thinking another method called something like "withConflictState" which would ensure conflict state is not null and then call `_finishConflictedMerge`. That way the side-effect-i-ness would be contained to a single clear function and `_finishConflictedMerge` could remain more functional. 🤔